### PR TITLE
Initialize segments of every builtin and add included field

### PR DESCRIPTION
**Pull request by Jrigada**
# Initialize segments of every builtin and add included field

## Description

We had to make some changes to the security.rs tests because now for them to work as intended we need to have the BuiltinRunner::finalize_segments() implemented. That is currently in another PR and we will revert the changes to this tests after that is merged. 
https://github.com/lambdaclass/cairo-rs/pull/542

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/vm/runners/cairo_runner.rs [hunk 1]
```diff
@@ -97,74 +97,68 @@ impl CairoRunner {
...
+        let mut difference = vec![];
+        for i in builtin_ordered_list {
+            if self.program.builtins.clone().contains(&i) {
+                difference.push(i);
             }
+        }
+        builtin_runners.push((
+            "output".to_string(),
+            OutputBuiltinRunner::new(difference.contains(&"output".to_string())).into(),
+        ));
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:111]:
```suggestion
        builtin_runners.push((
            "output".to_string(),
            OutputBuiltinRunner::new(self.program.builtins.contains(&"output".to_string())).into(),
        ));
```

File: src/vm/runners/cairo_runner.rs [hunk 2]
```diff
@@ -97,74 +97,68 @@ impl CairoRunner {
...
-                    )
+        builtin_runners.push((
+            "pedersen".to_string(),
+            HashBuiltinRunner::new(
+                if let Some(instance_def) = self.layout.builtins.pedersen.as_ref() {
+                    instance_def.ratio
+                } else {
+                    0
+                },
+                difference.contains(&"pedersen".to_string()),
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:121]:
```suggestion
                self.program.builtins.contains(&"pedersen".to_string()),
```

File: src/vm/runners/cairo_runner.rs [hunk 3]
```diff
@@ -97,74 +97,68 @@ impl CairoRunner {
...
+            .into(),
+        ));
+
+        if let Some(instance_def) = self.layout.builtins.range_check.as_ref() {
+            builtin_runners.push((
+                "range_check".to_string(),
+                RangeCheckBuiltinRunner::new(
+                    instance_def.ratio,
+                    instance_def.n_parts,
+                    difference.contains(&"range_check".to_string()),
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:132]:
```suggestion
                    self.program.builtins.contains(&"range_check".to_string()),
```

File: src/vm/runners/cairo_runner.rs [hunk 4]
```diff
@@ -97,74 +97,68 @@ impl CairoRunner {
...
-                ));
-            }
+            ));
         }
+
+        builtin_runners.push((
+            "bitwise".to_string(),
+            BitwiseBuiltinRunner::new(
+                self.layout.builtins.bitwise.as_ref(),
+                difference.contains(&"bitwise".to_string()),
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:148]:
```suggestion
                self.program.builtins.contains(&"bitwise".to_string()),
```

File: src/vm/runners/cairo_runner.rs [hunk 5]
```diff
@@ -97,74 +97,68 @@ impl CairoRunner {
...
+            )
+            .into(),
+        ));
+
+        builtin_runners.push((
+            "ec_op".to_string(),
+            EcOpBuiltinRunner::new(
+                self.layout.builtins.ec_op.as_ref(),
+                difference.contains(&"ec_op".to_string()),
```
Comment (fmoletta) [inline src/vm/runners/cairo_runner.rs:157]:
```suggestion
                self.program.builtinscontains(&"ec_op".to_string()),
```

**Review by fmoletta — CHANGES_REQUESTED** [review]

**Comment by Oppen** [comment]
Why do `InstanceDef`s become `Option<_>`? What's the use of `included`?

**Review by Oppen — CHANGES_REQUESTED** [review]

File: src/vm/runners/builtin_runner/bitwise.rs [hunk 6]
```diff
@@ -14,25 +14,31 @@ use crate::vm::errors::runner_errors::RunnerError;
...
 use crate::vm::vm_memory::memory_segments::MemorySegmentManager;
-
+#[derive(Debug)]
 pub struct BitwiseBuiltinRunner {
     _ratio: u32,
     pub base: isize,
     pub(crate) cells_per_instance: u32,
     _n_input_cells: u32,
-    bitwise_builtin: BitwiseInstanceDef,
+    bitwise_builtin: Option<BitwiseInstanceDef>,
```
[the remark this answers is not in the document]
Comment (Jrigada) [reply src/vm/runners/builtin_runner/bitwise.rs:25]: I am still trying to grasp this idea. I do not understand it that well, it seems to me that I would still have a lot of Results due to the checks of the ZeroDivisionErrors. We would need to look into where we need to use this instance def with none values.

File: docs/hint_processor/builtin_hint_processor/README.md [hunk 7]
```diff
@@ -50,7 +50,6 @@ fn print_a_hint(
     constants: &HashMap<String, BigInt>,
 ) -> Result<(), VirtualMachineError> {
     let a = get_integer_from_var_name("a", vm, ids_data, ap_tracking)?;
-    println!("{}", a);
```
Comment (fmoletta) [inline docs/hint_processor/builtin_hint_processor/README.md:53]: Why did you remove this line?

File: src/vm/runners/builtin_runner/bitwise.rs [hunk 8]
```diff
@@ -147,7 +154,11 @@ impl BitwiseBuiltinRunner {
         }
     }
 
-    pub fn get_used_diluted_check_units(&self, diluted_spacing: u32, diluted_n_bits: u32) -> usize {
+    pub fn get_used_diluted_check_units(
+        &self,
+        diluted_spacing: u32,
+        diluted_n_bits: u32,
+    ) -> Result<usize, RunnerError> {
```
Comment (fmoletta) [inline src/vm/runners/builtin_runner/bitwise.rs:161]: Remove this result

File: src/vm/runners/builtin_runner/mod.rs [hunk 9]
```diff
@@ -166,12 +167,16 @@ impl BuiltinRunner {
         }
     }
 
-    pub fn get_used_diluted_check_units(&self, diluted_spacing: u32, diluted_n_bits: u32) -> usize {
+    pub fn get_used_diluted_check_units(
+        &self,
+        diluted_spacing: u32,
+        diluted_n_bits: u32,
+    ) -> Result<usize, RunnerError> {
```
Comment (fmoletta) [inline src/vm/runners/builtin_runner/mod.rs:174]: Remove this result too

**Review by Juan-M-V — APPROVED** [review]
lgtm

**Review by fmoletta — APPROVED** [review]

**Review by juanbono — APPROVED** [review]
LGTM ser!

**Review by Oppen — APPROVED** [review]

**Pull request merged**